cli: TestPartialZip failed

cli.TestPartialZip [failed](https://mesolite.cluster.engflow.com/invocations/default/ab9fddae-d23e-4298-8f30-35d3ccac2516?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on release-25.4 @ [ab16afb782942bdb2f306440c5394bcc1eabfe22](https://github.com/cockroachdb/cockroach/commits/ab16afb782942bdb2f306440c5394bcc1eabfe22):


```
          zip
        output didn't match expected:
          @@ -41,10 +41,43 @@
           [node 1] writing ranges... writing JSON output: debug/nodes/1/ranges.json... done
           [node 2] skipping excluded node
           [node 3] node status... writing JSON output: debug/nodes/3/status.json... done
           [node 3] using SQL connection URL: postgresql://...
           <dumping SQL tables>
          + done
          +<dumping SQL tables>
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          + done
          +<dumping SQL tables>
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          + done
          +<dumping SQL tables>
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          + done
          +<dumping SQL tables>
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          + done
          +<dumping SQL tables>
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          + done
          +<dumping SQL tables>
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          + done
          +<dumping SQL tables>
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          + done
          +<dumping SQL tables>
          +NOTICE: No license is installed. Throttling will begin after 2025-11-23 09:09:13 +0000 UTC unless a license is installed before then.
          + done
          +<dumping SQL tables>
           [node 3] requesting data for debug/nodes/3/details... received response... writing JSON output: debug/nodes/3/details.json... done
           [node 3] requesting data for debug/nodes/3/gossip... received response... writing JSON output: debug/nodes/3/gossip.json... done
           [node 3] requesting stacks... received response... writing binary output: debug/nodes/3/stacks.txt... done
           [node 3] requesting stacks with labels... received response... writing binary output: debug/nodes/3/stacks_with_labels.txt... done
           [node 3] requesting heap profile... received response... writing binary output: debug/nodes/3/heap.pprof... done
    panic.go:635: -- test log scope end --
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestPartialZip1265502426
--- FAIL: TestPartialZip (33.65s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #154424 cli: TestPartialZip failed [C-test-failure O-robot T-observability branch-master release-blocker]
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server @cockroachdb/obs-india-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPartialZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-56995

Epic CRDB-56325

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -698,6 +698,10 @@ func eraseNonDeterministicZipOutput(out string) string {
 	re = regexp.MustCompile(`(?m)^\[node \d+\] \d+ execution traces found$`)
 	out = re.ReplaceAllString(out, `[node ?] ? execution traces found`)
 
+	// Remove license-related NOTICE messages that may appear intermittently.
+	re = regexp.MustCompile(`(?m)^NOTICE: No license is installed.*\n?`)
+	out = re.ReplaceAllString(out, ``)
+
 	return out
 }
 

